src/blocks/keyboard_layout.rs

```diff
@@ -318,3 +330,93 @@ trait LocaleBusInterface {
     #[dbus_proxy(property, name = "X11Variant")]
     fn variant(&self) -> zbus::Result<String>;
 }
+
+#[dbus_proxy(
+    interface = "ru.gentoo.kbdd",
+    default_service = "ru.gentoo.KbddService",
+    default_path = "/ru/gentoo/KbddService"
+)]
+trait KbddBusInterface {
+    #[dbus_proxy(signal, name = "layoutNameChanged")]
+    fn layout_updated(&self, layout: String) -> zbus::Result<()>;
+
+    #[dbus_proxy(name = "getCurrentLayout")]
+    fn current_layout_index(&self) -> zbus::Result<u32>;
+
+    #[dbus_proxy(name = "getLayoutName")]
+    fn current_layout(&self, layout_id: u32) -> zbus::Result<String>;
+}
+
+struct KbddBus {
+    stream: layoutNameChangedStream<'static>,
+    layout: String,
+    variant: Option<String>,
+    re_layout: Regex,
+}
+
+impl KbddBus {
+    async fn new() -> Result<Self> {
+        let conn = new_dbus_connection().await?;
+        let proxy = KbddBusInterfaceProxy::builder(&conn)
+            .cache_properties(zbus::CacheProperties::No)
+            .build()
+            .await
+            .error("Failed to create KbddBusInterfaceProxy")?;
+        let stream = proxy
+            .receive_layout_updated()
+            .await
+            .error("Failed to monitor kbdd interface")?;
+        let layout_index = proxy
+            .current_layout_index()
+            .await
+            .error("Failed to get current layout index from kbdd")?;
+        let current_layout = proxy
+            .current_layout(layout_index)
+            .await
+            .error("Failed to get current layout from kbdd")?;
+        let re_layout = Regex::new(r#"(.*) \((.*)\)"#).unwrap(); //to split layout from variant in string like 'English (US)'
+        let (layout, variant) = Self::parse_kbdd_layout(&re_layout, &current_layout);
+        Ok(Self {
+            stream,
+            layout,
+            variant,
+            re_layout,
+        })
+    }
+
+    fn parse_kbdd_layout(re_layout: &Regex, layout_name: &str) -> (String, Option<String>) {
```

Comment:
Similarly, this function can return `Info`

Reply:
Oh, and by the way, it looks like this function does exactly the same thing as `parse_sway_output()`?

Reply:
Thanks, good catch, I guess I had to browse the surrounding code a little bit more.